test/compiler/codegen.jl

```diff
@@ -407,7 +407,7 @@ function g_dict_hash_alloc()
 end
 # Warm up
 f_dict_hash_alloc(); g_dict_hash_alloc();
-@test (@allocated f_dict_hash_alloc()) == (@allocated g_dict_hash_alloc())
+@test abs((@allocated f_dict_hash_alloc()) / (@allocated g_dict_hash_alloc()) - 1) < 0.1 # less that 10% difference
```

Comment:
This test is flaky, and causes CI failures: #56748

What is the reasoning behind `0.1` as a cutoff.

Here's a recent CI failure: https://buildkite.com/julialang/julia-master/builds/44575#0194e891-29ce-4ae2-a22c-81cec7ea67d7/750-1077